virttest/qemu_vm.py

```diff
@@ -530,12 +530,26 @@ def add_name(name):
             return " -name '%s'" % name
 
         def process_sandbox(devices, action):
+            sandbox_option = ""
+            if not devices.has_option("sandbox"):
+                return sandbox_option
             if action == "add":
-                if devices.has_option("sandbox"):
-                    return " -sandbox on "
+                sandbox_option = " -sandbox on"
             elif action == "rem":
-                if devices.has_option("sandbox"):
-                    return " -sandbox off "
+                sandbox_option = " -sandbox off"
+            elevateprivileges = params.get("qemu_sandbox_elevateprivileges")
+            if elevateprivileges is not None:
+                sandbox_option += ",elevateprivileges=" + elevateprivileges
+            obsolete = params.get("qemu_sandbox_obsolete")
+            if obsolete is not None:
+                sandbox_option += ",obsolete=" + obsolete
+            resourcecontrol = params.get("qemu_sandbox_resourcecontrol")
+            if resourcecontrol is not None:
+                sandbox_option += ",resourcecontrol=" + resourcecontrol
+            spawn = params.get("qemu_sandbox_spawn")
+            if spawn is not None:
+                sandbox_option += ",spawn=" + spawn
```

Comment:
```suggestion
            for key, val in {"elevateprivileges": params.get("qemu_sandbox_elevateprivileges"),
                             "obsolete": params.get("qemu_sandbox_obsolete"), 
                             "resourcecontrol": params.get("qemu_sandbox_resourcecontrol"),
                             "spawn": params.get("qemu_sandbox_spawn")}:
                if val is not None:
                    sandbox_option += f",{key}=" + val
```

Reply:
@YongxueHong I face the following error:
```bash
[stdlog] 2023-09-12 03:53:04,403 avocado.test ERROR|   File "/usr/local/lib/python3.9/site-packages/avocado_framework_plugin_vt-100.1-py3.9.egg/virttest/qemu_vm.py", line 540, in process_sandbox
[stdlog] 2023-09-12 03:53:04,403 avocado.test ERROR|     for opt, val in {"elevateprivileges": params.get("qemu_sandbox_elevateprivileges"),
[stdlog] 2023-09-12 03:53:04,403 avocado.test ERROR| ValueError: too many values to unpack (expected 2)
```

Reply:
Hi @mcasquer 
My bad, I missed something, I suggest that we can use a `dict` object to store the option and the related value, and then we get them iteratively.
Like:
```
_params = {"elevateprivileges": params.get("qemu_sandbox_elevateprivileges"), 
        "obsolete": params.get("qemu_sandbox_obsolete"),  
        "resourcecontrol": params.get("qemu_sandbox_resourcecontrol")}
for opt, val in _params.items():
```

Reply:
@YongxueHong thanks for the suggestion, now the cases pass as expected:
```bash
 (1/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.numa_prealloc_threads.q35: STARTED
 (1/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.numa_prealloc_threads.q35: PASS (639.44 s)
 (2/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.numa_prealloc_threads.sandbox_off.q35: STARTED
 (2/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.numa_prealloc_threads.sandbox_off.q35: PASS (639.42 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 
```